Framework/CurveFitting/src/Functions/Meier.cpp

```diff
@@ -0,0 +1,222 @@
+// Mantid Repository : https: // github.com/mantidproject/mantid
+//
+// Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+// NScD Oak Ridge National Laboratory, European Spallation Source,
+// Institut Laue - Langevin &CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier : GPL - 3.0 +
+#include "MantidCurveFitting/Functions/Meier.h"
+#include "MantidAPI/FunctionFactory.h"
+#include <cmath>
+#include <valarray>
+
+namespace Mantid::CurveFitting::Functions {
+
+using namespace CurveFitting;
+
+using namespace Kernel;
+
+using namespace API;
+
+DECLARE_FUNCTION(Meier)
+
+namespace {
+bool isMultipleOf05(double value) {
+  double quantum = 0.01;
+  double threshold = quantum / 2;
+  return std::fmod(value, 0.5) < threshold;
+}
+} // namespace
+
+void Meier::init() {
+  declareParameter("A0", 0.5, "Amplitude");
+  declareParameter("FreqD", 0.01, "Angular Frequency due to dipolar coupling (MHz)");
+  declareParameter("FreqQ", 0.05,
+                   "Angular Frequency due to quadrupole interaction of the nuclear spin (MHz) due to a field gradient"
+                   "exerted by the presence of the muon");
+  declareParameter("Sigma", 0.2, "Gaussian decay rate");
+  declareParameter("Lambda", 0.1, "Exponential decay rate");
+  // J, Total angular momentum quanutm number
+  declareAttribute("Spin", API::IFunction::Attribute(3.5));
+}
+
+void Meier::function1D(double *out, const double *xValues, const size_t nData) const {
+  const double J = getAttribute("Spin").asDouble();
+  if (!isMultipleOf05(J)) {
```

Comment:
Can we use an attribute validator?https://docs.mantidproject.org/nightly/tutorials/extending_mantid_with_python/python_fit_functions/03_attributes.html#attributes

Reply:
Ok, I will add the validator.

Ok, I will refactor them